src/components/NcAppSidebar/NcAppSidebar.vue

```diff
@@ -366,155 +370,237 @@ A working alternative would be using an icon together with an `aria-label`:
-						:style="{
-							backgroundImage: `url(${background})`
-						}"
-						tabindex="0"
-						@click="onFigureClick"
-						@keydown.enter="onFigureClick">
-						<slot class="app-sidebar-header__background" name="header" />
-					</div>
+	<!-- This is in most cases NcContent -->
+	<div class="content-wrapper">
+		<!-- The main content - In most cases NcAppContent -->
+		<div>
+			<NcButton @click.prevent="showSidebar = !showSidebar">
+				Toggle sidebar
+			</NcButton>
+		</div>
+		<!-- The sidebar -->
+		<NcAppSidebar
+			:open.sync="showSidebar"
+			name="cat-picture.jpg"
+			subname="last edited 3 weeks ago">
+			<NcAppSidebarTab name="Settings" id="settings-tab">
+				<template #icon>
+					<Cog :size="20" />
+				</template>
+				Single tab content
+			</NcAppSidebarTab>
+		</NcAppSidebar>
+	</div>
+</template>
+<script>
+import Cog from 'vue-material-design-icons/Cog'
 
-					<!-- sidebar details -->
-					<div v-if="!empty"
-						:class="{
-							'app-sidebar-header__desc--with-tertiary-action': canStar || $slots['tertiary-actions'],
-							'app-sidebar-header__desc--editable': nameEditable && !subname,
-							'app-sidebar-header__desc--with-subname--editable': nameEditable && subname,
-							'app-sidebar-header__desc--without-actions': !$slots['secondary-actions'],
-						}"
-						class="app-sidebar-header__desc">
-						<!-- favourite icon -->
-						<div v-if="canStar || $slots['tertiary-actions']" class="app-sidebar-header__tertiary-actions">
-							<slot name="tertiary-actions">
-								<NcButton v-if="canStar"
-									:aria-label="favoriteTranslated"
-									:pressed="isStarred"
-									class="app-sidebar-header__star"
-									type="secondary"
-									@click.prevent="toggleStarred">
-									<template #icon>
-										<NcLoadingIcon v-if="starLoading" />
-										<Star v-else-if="isStarred" :size="20" />
-										<StarOutline v-else :size="20" />
-									</template>
-								</NcButton>
-							</slot>
+export default {
+	components: {
+		Cog,
+	},
+	data() {
+		return {
+			showSidebar: true,
+		}
+	},
+}
+</script>
+<style scoped>
+/* This styles just mock NcContent and NcAppContent */
+.content-wrapper {
+	position: relative;
+	/* Just to prevent jumping when the sidebar is hidden */
+	min-height: 360px;
+}
+
+.main-content {
+	position: absolute;
+	height: 100%;
+	width: 100%;
+}
+
+/* Fix styles on this style guide page */
+@media only screen and (max-width: 512px) {
+	:deep(aside) {
+		width: calc(100vw - 64px) !important;
+	}
+}
+</style>
+```
+</docs>
+
+<template>
+	<Fragment>
+		<!-- With vue3 we can move this inside the transition, but vue2 does not allow v-show in transition -->
+		<NcButton v-show="!open"
+			:aria-label="t('Open sidebar')"
+			class="app-sidebar__toggle"
+			:class="toggleClasses"
+			type="tertiary"
+			@click="$emit('update:open', true)">
+			<template #icon>
+				<IconDockRight :size="20" />
+			</template>
+		</NcButton>
+		<transition appear
```

Comment:
This is a small breaking change
- Before `<aside>` through `<Transition>` was actually a root element receiving all passed attrs such as classes, styles or `data-cy`
- After it is a fragment, no attributes are passed.

So we need to bind `$attrs` and `$listeners`, and add `inheritAttrs: false`